openapi/src/parts/controlled_azure_batch_pool.yaml

```diff
@@ -154,14 +154,21 @@ components:
 
     AzureBatchPoolUserAssignedIdentity:
       type: object
-      description: User assigned identity associated with a batch pool
-      required: [ name, resourceGroupName ]
+      description: User assigned identity associated with a batch pool.
       properties:
         name:
-          description: The name of user assigned identity.
+          description: |
+            The name of the user assigned identity. This parameter is mutually exclusive with clientId.
```

Comment:
Given `name` and `clientId` are mutually exclusive, how would a caller decide to pass one or the other? Are there circumstances where `name` is required and others where `clientId` is required?

Reply:
It depends on what a caller has. Both options will give the same result. But Azure Batch SDK accept name of managed identity. So, in case a caller provide name it is straightforward to call corresponding API. Otherwise, we need to look up for it using clientId. So, it is an additional call to Azure.